src/components/select/multi-select/multi-select.component.tsx

```diff
@@ -455,56 +468,115 @@ export const MultiSelect = React.forwardRef<HTMLInputElement, MultiSelectProps>(
     function handleTextboxClick(event: React.MouseEvent<HTMLInputElement>) {
       isMouseDownReported.current = false;
 
-      onClick?.(event);
-
-      if (openOnFocus && isOpenedByFocus) {
-        setIsOpenedByFocus(false);
-        return;
+      if (onClick) {
+        onClick(event);
       }
 
-      if (open) {
-        setTextValue("");
-        setOpen(false);
-      } else {
+      /* istanbul ignore else */
+      if (!openOnFocus || (openOnFocus && !isOpenedByFocus.current)) {
+        if (isOpen) {
+          setFilterText("");
+          setOpenState(false);
+          return;
+        }
+
         onOpen?.();
-        setOpen(true);
+
+        setOpenState(true);
+      } else {
+        // This is tested in Playwright.
+        // This line of code is triggered when a user initially clicks on the input when `openOnFocus` is true.
+        isOpenedByFocus.current = false;
       }
     }
 
+    function handleDropdownIconClick(
+      event: React.MouseEvent<HTMLElement> | React.KeyboardEvent<HTMLElement>,
+    ) {
+      isMouseDownReported.current = false;
+
+      if (onClick) {
+        onClick(event as React.MouseEvent<HTMLInputElement>);
+      }
+
+      setOpenState((isAlreadyOpen) => {
+        if (isAlreadyOpen) {
+          setFilterText("");
+          return false;
+        }
+
+        if (onOpen) {
+          onOpen();
+        }
+
+        return true;
+      });
+    }
+
     function handleTextboxBlur(event: React.FocusEvent<HTMLInputElement>) {
       isMouseDownOnInput.current = false;
 
       if (isMouseDownReported.current) {
         return;
       }
 
-      onBlur?.(event);
+      isInputFocused.current = false;
 
-      setTextValue("");
-      setOpen(false);
+      if (onBlur) {
+        onBlur(event);
+      }
     }
 
-    function handleTextboxMouseDown() {
+    function handleTextboxMouseDown(event: React.MouseEvent<HTMLElement>) {
       isMouseDownReported.current = true;
-      isMouseDownOnInput.current = true;
+
+      if ((event.target as HTMLInputElement).dataset.element === "input") {
+        isMouseDownOnInput.current = true;
+      }
     }
 
     function handleListMouseDown() {
       isMouseDownReported.current = true;
     }
 
     function handleTextboxFocus(event: React.FocusEvent<HTMLInputElement>) {
-      onFocus?.(event);
+      const triggerFocus = () => onFocus?.(event);
 
       if (openOnFocus) {
-        if (open) return;
-
-        onOpen?.();
-
-        if (isMouseDownOnInput.current) {
-          setIsOpenedByFocus(true);
+        if (focusTimer.current) {
+          clearTimeout(focusTimer.current);
         }
-        setOpen(true);
+
+        // we need to use a timeout here as there is a race condition when rendered in a modal
+        // whereby the select list isn't visible when the select is auto focused straight away
+        focusTimer.current = setTimeout(() => {
```

Comment:
I've sent Dan some potential tweaks to this so worth waiting for him to investigate if they're viable before reviewing